x-pack/plugins/maps/public/shared/layers/styles/vector_style.js

```diff
@@ -127,6 +130,52 @@ export class VectorStyle extends AbstractStyle {
     };
   }
 
+  pluckStyleMetaFromSourceDataRequest(sourceDataRequest) {
+    const features = _.get(sourceDataRequest.getData(), 'features', []);
+    if (features.length === 0) {
+      return {};
+    }
+
+    const scaledFields = this._getDynamicPropertiesArray()
+      .map(({ options }) => {
+        return {
+          name: options.field.name,
+          min: Infinity,
+          max: -Infinity
+        };
+      });
+
+    let isPointsOnly = true;
```

Comment:
fwiw a nested for-loop will read easier than this nesting of lexical scopes.

a functional approach also creates a new function object for each feature to loop over the scaled-loops and adds the extra indirection of the extra function-calls.

tight loops, like this one,  are more readable and less crufty wrt object allocation and function call indirection when keeping execution local to the stack-frame.

e.g.
```
    for (let i = 0; i < features.length; i++) {
      const feature = features[i];
      if (feature.geometry.type !== 'Point') {
        isPointsOnly = false;
      }
      for (let j = 0; j < scaledFields.length; j++) {
        const scaledField = scaledFields[j];
        const newValue = parseFloat(feature.properties[scaledField.name]);
        if (!isNaN(newValue)) {
          scaledField.min = Math.min(scaledField.min, newValue);
          scaledField.max = Math.max(scaledField.max, newValue);
        }
      }
    }
```